staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factory.go

```diff
@@ -176,15 +178,26 @@ func NewSharedInformerFactoryWithOptions(client {{.clientSetInterface|raw}}, def
 
 // Start initializes all requested informers.
 func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
-  f.lock.Lock()
-  defer f.lock.Unlock()
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	for informerType, informer := range f.informers {
+		if !f.startedInformers[informerType] {
+			f.wg.Add(1)
+			informer := informer
+			go func() {
+				defer f.wg.Done()
+				informer.Run(stopCh)
+			}()
+			f.startedInformers[informerType] = true
+		}
+	}
+}
 
-  for informerType, informer := range f.informers {
-    if !f.startedInformers[informerType] {
-      go informer.Run(stopCh)
-      f.startedInformers[informerType] = true
-    }
-  }
+// WaitForShutdown waits for the stop of the goroutines started by one or more
+// previous Start calls.
```

Comment:
This is the same as:
```
factory := NewSharedInformerFactory(....)
_ := factory.InformerFor(gvk1)
factory.Start(stop1)
_ := factory.InformerFor(gvk2)
factory.Start(stop2)
close(stop1)
factory.WaitForShutdown()
```

In other words, it will hang until some other goroutine closes `stop2`.

I need to change the description to "WaitForShutdown waits for the factory to have no goroutines running anymore."